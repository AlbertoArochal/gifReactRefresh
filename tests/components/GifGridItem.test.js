import {GifGridItem} from "../../src/components/index.js";
import {render} from "@testing-library/react";
import {screen} from "@testing-library/dom";

describe('Pruebas en gifGridItem', () => {
    const title = 'One Punch';
    const url = 'https://sm.ign.com/t/ign_latam/screenshot/default/chechi_ewwz.1280.jpg'
    test("debe hacer match con el snapshot", () => {


        const {container} = render(<GifGridItem title={title} url={url} />)
        expect(container).toMatchSnapshot();


        }
    )
    test(
        "debe mostrart la imagen", () => {
            render(<GifGridItem title={title} url={url} />)

            expect(screen.getByRole('img').src).toBe(url)
        }
    )
    test(
        "debe mostrar el titulo", () => {
            render(<GifGridItem title={title} url={url} />)
        }
    )
});