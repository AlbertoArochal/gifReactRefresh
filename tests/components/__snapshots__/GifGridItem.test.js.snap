// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en gifGridItem debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="One Punch"
      src="https://sm.ign.com/t/ign_latam/screenshot/default/chechi_ewwz.1280.jpg"
    />
    <p>
      One Punch
    </p>
  </div>
</div>
`;
